.github/workflows/shared_check_maintainer.yml

```diff
@@ -30,41 +26,27 @@ jobs:
     outputs:
       is_core_team: ${{ steps.check_if_actor_is_maintainer.outputs.result }}
     steps:
-      - uses: actions/checkout@v4
```

Comment:
@poteto have you thought about using `action/checkout` to checkout the main branch? It has options for selecting the repo and branch, so it could be:

```yaml
steps:
  - uses action/checkout@v4
  - with:
    ref: main
    repository: facebook/react
```

Reply:
I'm not sure if it's a good idea, from what I read it's not a good idea to checkout any code from something triggered via `pull_request_target` since it has write access to the target repo and access to secrets. 

Always fetching it from `main` seems like the safer choice, albeit less convenient